mock/py/mockbuild/plugins/rpmautospec.py

```diff
@@ -0,0 +1,117 @@
+        self.log.info("rpmautospec: initialized")
+
+    @contextmanager
+    def as_root_user(self):
+        try:
+            self.buildroot.uid_manager.becomeUser(0, 0)
+            yield
+        finally:
+            self.buildroot.uid_manager.restorePrivs()
+
+    @traceLog()
+    def attempt_process_distgit(
+        self,
+        host_chroot_spec: Union[Path, str],
+        host_chroot_sources: Optional[Union[Path, str]],
+    ) -> None:
+        # Set up variables and check prerequisites.
+        if not host_chroot_sources:
+            self.log.debug("Sources not specified, skipping rpmautospec preprocessing.")
+            return
+
+        host_chroot_spec = Path(host_chroot_spec)
+        host_chroot_sources = Path(host_chroot_sources)
+        if not host_chroot_sources.is_dir():
+            self.log.debug(
+                "Sources not a directory, skipping rpmautospec preprocessing."
+            )
+            return
+
+        distgit_git_dir = host_chroot_sources / ".git"
+        if not distgit_git_dir.is_dir():
+            self.log.debug(
+                "Sources is not a git repository, skipping rpmautospec preprocessing."
+            )
+            return
+
+        host_chroot_sources_spec = host_chroot_sources / host_chroot_spec.name
+        if not host_chroot_sources_spec.is_file():
+            self.log.debug(
+                "Sources doesn’t contain spec file, skipping rpmautospec preprocessing."
+            )
+            return
+
+        with host_chroot_spec.open("rb") as spec, host_chroot_sources_spec.open(
+            "rb"
+        ) as sources_spec:
+            if spec.read() != sources_spec.read():
+                self.log.warning(
+                    "Spec file inside and outside sources are different, skipping rpmautospec"
+                    " preprocessing."
+                )
+                return
+
+        if not specfile_uses_rpmautospec(host_chroot_sources_spec):
+            self.log_debug(
+                "Spec file doesn’t use rpmautospec, skipping rpmautospec preprocessing."
+            )
+            return
+
+        # Install the `rpmautospec` command line tool into the build root.
+        if self.opts.get("requires", None):
+           self.buildroot.pkg_manager.install_as_root(*self.opts["requires"], check=True)
+
+        # Get paths inside the chroot by chopping off the leading paths
+        rootdir_prefix = self.buildroot.make_chroot_path()
+        chroot_spec = host_chroot_spec.replace(rootdir_prefix, "")
+        chroot_sources = host_chroot_sources.replace(rootdir_prefix, "")
+        chroot_sources_spec = host_chroot_sources_spec.replace(rootdir_prefix, "")
+
+        # Call subprocess to perform the specfile rewrite
+        command = self.get("cmd_base")
+        command += ["--input", chroot_sources_spec]
+        command += ["--output", chroot_spec]
+        command += ["--git-dir", chroot_sources]
```

Comment:
We could roll these 4 statements into one, but that’s just me being obsessive. 😉

Reply:
I didn't realize that command already existed. Handy. I'll fix that.

In general, I prefer the style of having them as separate lines specifically because it makes it easier to review patches in the future if we end up adding new arguments.